File: js/ejer57.js
//validar contraseña
function validarContraseña() {
    const contraseñaInput = document.getElementById("contraseña");
    const mensaje = document.getElementById("mensaje");

    if (!contraseñaInput || !mensaje) {
        console.error("No se encontraron los elementos #contraseña o #mensaje.");
        return;
    }

    const contraseña = contraseñaInput.value;
    if (contraseña.length < 8) {
        mensaje.textContent = "La contraseña debe tener al menos 8 caracteres.";
        mensaje.style.color = "red";
    } else {
        mensaje.textContent = "Contraseña válida.";
        mensaje.style.color = "green";
    }
}

window.addEventListener("DOMContentLoaded", function () {
    const contraseñaInput = document.getElementById("contraseña");
    if (contraseñaInput) {
        contraseñaInput.addEventListener("input", validarContraseña);
        contraseñaInput.addEventListener("keydown", function (event) {
            if (event.key === "Enter") {
                event.preventDefault();
                validarContraseña();
            }
        });
    }
});

File: python/ejer57.py
#validar contraseña
def validar_contraseña(contraseña):
    if len(contraseña) < 8:
        return False
    if not any(char.isupper() for char in contraseña):
        return False
    if not any(char.islower() for char in contraseña):
        return False
    if not any(char.isdigit() for char in contraseña):
        return False
    if not any(char in "!@#$%^&*()-_=+[]{}|;:'\",.<>?/" for char in contraseña):
        return False
    return True
contraseña = input("Ingrese una contraseña: ")
if validar_contraseña(contraseña):
    print("Contraseña válida.")
else:    print("Contraseña inválida. Debe tener al menos 8 caracteres, incluir mayúsculas, minúsculas, números y caracteres especiales.")

